static-analysis-plugin/plugin/src/main/groovy/com/novoda/staticanalysis/PmdConfigurator.groovy

```diff
@@ -0,0 +1,90 @@
+package com.novoda.staticanalysis
+
+import com.novoda.staticanalysis.PmdViolationsEvaluator.PmdViolation
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.plugins.quality.Pmd
+import org.gradle.api.plugins.quality.PmdExtension
+import org.gradle.internal.logging.ConsoleRenderer
+
+class PmdConfigurator {
+
+    void configure(Project project, Violations violations, StaticAnalysisExtension extension, Task evaluateViolations) {
+        extension.ext.pmd = { Closure config ->
+            project.apply plugin: 'pmd'
+            List<String> excludes = []
+            configureExtension(project.extensions.findByType(PmdExtension), excludes, config)
+            project.afterEvaluate {
+                boolean isAndroidApp = project.plugins.hasPlugin('com.android.application')
+                boolean isAndroidLib = project.plugins.hasPlugin('com.android.library')
+                if (isAndroidApp || isAndroidLib) {
+                    def variants = isAndroidApp ? project.android.applicationVariants : project.android.libraryVariants
+                    configureAndroid(project, variants)
+                }
+                project.tasks.withType(Pmd) { pmd ->
+                    configureTask(pmd, violations, excludes)
+                    evaluateViolations.dependsOn pmd
+                }
+            }
+        }
+    }
+
+    private Object configureExtension(PmdExtension extension, List<String> excludes, Closure config) {
+        extension.with {
+            toolVersion = '5.5.1'
```

Comment:
not sure @florianmski. on one hand we want _close-to-none_ setup, on the other we seek flexibility: I figured that we can always update the plugin to expose such properties in the future when needed (YAGNI & all that stuff).